src/components/file/File.js

```diff
@@ -783,6 +783,42 @@ export default class FileComponent extends BaseComponent {
         this.uploadStatusList.appendChild(uploadStatus);
 
         if (fileUpload.status !== 'error') {
+          // Track uploads in progress.
+          if (fileService.uploadsInProgress === undefined) {
+            const cssClass = 'uploads-in-progress';
+            var submitButton = this.root.element
+                .querySelector('.formio-component-submit')
+                .querySelector('button');
+            var array = new Array();
+            fileService.uploadsInProgress = {
+              array: array,
+              add: function(item) {
+                if (cssClass && (array.length === 0)) {
+                  submitButton.classList.add(cssClass);
+                }
+                array.push(item);
+              },
+              remove: function(item) {
+                array.splice(array.indexOf(item), 1);
+                if (cssClass && (array.length === 0)) {
+                  submitButton.classList.remove(cssClass);
+                }
+              },
+              report: function() {
+                const n = array.length;
+                var msg = `Uploads in progress: ${n}`;
+                if (n>0) {
+                  const keys = array.map(x => x.component.key);
+                  msg += ` (${keys.join(', ')})`;
+                }
+                console.log(msg);
+              }
+            };
+          }
+          const uploadsInProgress = fileService.uploadsInProgress;
+          uploadsInProgress.add(this);
```

Comment:
what if ```uploadsInProgress``` is null here?

Reply:
Shouldn't happen, at least I can't see how.

In the patch above, it is created before (first) use, and then never cleared.
if (fileService.uploadsInProgress === undefined) {
            fileService.uploadsInProgress = {...}
}

As it works as intended, I can't see how it could be null.

I'd be more worried about multiple forms being displayed simultaneously on the same page interfering with one another; that's something I haven't tried yet.

Maybe the initialization of uploadsInProgress should happen together with fileService, but I'm not familiar enough with those portions of source code. Having it all in one place seemed simpler for now, for me at least.

Reply:
ok, I see that now.